Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -170,8 +170,13 @@ class TopTabsViewController: UIViewController {
     
     func privateTabTapped() {
         delegate?.topTabsDidPressPrivateTab()
-        self.collectionView.reloadData()
-        scrollToCurrentTab(false)
+        UIView.animateWithDuration(0, animations: {
```

Comment:
How come the 0 duration animation block? This is probably causing the delay @varkor method in the bug. You should be able to just call the reload and scroll immediately without the animation block.